include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -193,6 +195,7 @@ class IExecutionProvider {
  private:
   const std::string type_;
   AllocatorMap allocators_;
+  MemoryInfoSet mem_info_set_;
   //It will be set when this object is registered to a session
```

Comment:
Comment explaining what should be in this and possibly how it relates to allocators_ would be good